aws/resource_aws_api_gateway_rest_api.go

```diff
@@ -335,6 +344,20 @@ func resourceAwsApiGatewayRestApiUpdateOperations(d *schema.ResourceData) []*api
 		}
 	}
 
+	if d.HasChange("endpoint_configuration.0.vpc_endpoint_ids") {
+		if v, ok := d.GetOk("endpoint_configuration"); ok && len(v.([]interface{})) > 0 {
+			m := v.([]interface{})[0].(map[string]interface{})
+			if m["types"].([]interface{})[0].(string) == "PRIVATE" {
+				operations = append(operations, &apigateway.PatchOperation{
+					Op:    aws.String("replace"),
```

Comment:
`replace` doesn't work for the IDs. Only `add` and `remove` are supported. See https://docs.aws.amazon.com/apigateway/latest/developerguide/associate-private-api-with-vpc-endpoint.html